### PR TITLE
Split user info and user credentials

### PR DESCRIPTION
**Pull request by djino97**
- Split DbUser, added new model DbUserCredentials.
- Reworked business logic, unit-tests.

### CODE REVIEW

File: src/UserService.Mappers/UserMapper.cs [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 namespace LT.DigitalOffice.UserService.Mappers
 {
     /// <summary>
     /// Represents mapper. Provides methods for converting an object of <see cref="DbUser"/> type into an object of <see cref="User"/> type according to some rule.
     /// </summary>
-    public class UserMapper : IMapper<DbUser, User>, IMapper<DbUser, string, object>, IMapper<UserCreateRequest, DbUser>, IMapper<EditUserRequest, DbUser>
+    public class UserMapper : IMapper<DbUser, User>, IMapper<DbUser, string, object>,
+        IMapper<UserCreateRequest, DbUser>, IMapper<EditUserRequest, DbUser>,
+        IMapper<EditUserRequest, string, DbUserCredentials>,
+        IMapper<UserCreateRequest, Guid, DbUserCredentials>
```
Comment (lalagvana) [inline src/UserService.Mappers/UserMapper.cs:17]: Maybe we should separate User mappers and UserCredentials mappers?

Comment (djino97) [reply src/UserService.Mappers/UserMapper.cs:17]: Maybe)

Comment (djino97) [reply src/UserService.Mappers/UserMapper.cs]: Fixed

**Review by lalagvana — APPROVED** [review]

File: src/UserService.Broker/Consumers/UserLoginConsumer.cs [hunk 2]
```diff
@@ -27,11 +27,12 @@ public async Task Consume(ConsumeContext<IUserCredentialsRequest> context)
 
         private object GetUserCredentials(IUserCredentialsRequest request)
         {
-            DbUser userCredentials = repository.GetUserByEmail(request.Email);
+            DbUser user = repository.GetUserByEmail(request.Email);
```
Comment (DarkFelas) [inline src/UserService.Broker/Consumers/UserLoginConsumer.cs:30]: email now in userCreds, rework this => remove dbUser

File: src/UserService.Business/EditUserCommand.cs [hunk 3]
```diff
@@ -10,27 +10,35 @@ namespace LT.DigitalOffice.UserService.Business
...
             this.repository = repository;
-            this.mapper = mapper;
+            this.mapperUser = mapperUser;
+            this.mapperUserCredentials = mapperUserCredentials;
         }
 
-        public bool Execute(EditUserRequest request)
+        public bool Execute(UserRequest request)
         {
+            string salt = "Exmple_salt";
```
Comment (DarkFelas) [inline src/UserService.Business/EditUserCommand.cs:32]: wtf??

File: src/UserService.Business/EditUserCommand.cs [hunk 4]
```diff
@@ -10,27 +10,35 @@ namespace LT.DigitalOffice.UserService.Business
...
             validator.ValidateAndThrow(request);
 
-            var user = mapper.Map(request);
+            var dbUser = mapperUser.Map(request);
+            var dbUserCredentials = mapperUserCredentials.Map(request);
+
+            dbUserCredentials.Salt = salt;
 
-            return repository.EditUser(user);
+            return repository.EditUser(dbUser) && repository.EditUserCredentials(dbUserCredentials);
```
Comment (DarkFelas) [inline src/UserService.Business/EditUserCommand.cs:41]: bad practice, what if something goes wrong in EditUser, should we save Credentials ?

File: src/UserService.Business/UserCreateCommand.cs [hunk 5]
```diff
@@ -1,36 +1,46 @@
...
 using LT.DigitalOffice.UserService.Data.Interfaces;
 using LT.DigitalOffice.UserService.Mappers.Interfaces;
 using LT.DigitalOffice.UserService.Models.Db;
 using LT.DigitalOffice.UserService.Models.Dto;
 using Microsoft.AspNetCore.Mvc;
+using System;
 
 namespace LT.DigitalOffice.UserService.Business
 {
     public class UserCreateCommand : IUserCreateCommand
```
Comment (DarkFelas) [inline src/UserService.Business/UserCreateCommand.cs:12]: rename => CreateUserCommand

File: src/UserService.Business/UserCreateCommand.cs [hunk 6]
```diff
@@ -1,36 +1,46 @@
...
-            var user = mapper.Map(request);
 
-            return repository.UserCreate(user);
+            var dbUser = mapperUser.Map(request);
+            var dBUserCredentials = mapperUserCredentials.Map(request);
+
+            dBUserCredentials.Id = Guid.NewGuid();
+            dBUserCredentials.UserId = dbUser.Id;
+
+            repository.CreateUserCredentials(dBUserCredentials);
```
Comment (DarkFelas) [inline src/UserService.Business/UserCreateCommand.cs:41]: first create user and only on success -> create credentials

File: src/UserService.Data/Interfaces/IUserRepository.cs [hunk 7]
```diff
@@ -36,5 +37,25 @@ public interface IUserRepository
         /// <param name="userEmail">Specified email of user.</param>
         /// <returns>User model.</returns>
         DbUser GetUserByEmail(string userEmail);
+
+        /// <summary>
+        /// Returns the user credentials with the specified user id from database.
+        /// </summary>
+        /// <param name="userId">Specified Id of user.</param>
+        /// <returns>User credentials model.</returns>
+        DbUserCredentials GetUserCredentialsById(Guid userId);
```
Comment (DarkFelas) [inline src/UserService.Data/Interfaces/IUserRepository.cs:46]: Move credentials methods to new repository

**Review by DarkFelas — CHANGES_REQUESTED** [review]

File: src/UserService.Broker/Consumers/UserLoginConsumer.cs (hunk 2 above)
Comment (djino97) [reply src/UserService.Broker/Consumers/UserLoginConsumer.cs]: Fixed

File: src/UserService.Business/EditUserCommand.cs (hunk 3 above)
Comment (djino97) [reply src/UserService.Business/EditUserCommand.cs]: Removed

File: src/UserService.Business/EditUserCommand.cs (hunk 4 above)
Comment (djino97) [reply src/UserService.Business/EditUserCommand.cs]: it was allowed in messages in slack

File: src/UserService.Business/UserCreateCommand.cs (hunk 5 above)
Comment (djino97) [reply src/UserService.Business/UserCreateCommand.cs]: Fixed

File: src/UserService.Data/Interfaces/IUserRepository.cs (hunk 7 above)
Comment (djino97) [reply src/UserService.Data/Interfaces/IUserRepository.cs]: Splitted repository

Comment (djino97) [reply src/UserService.Data/Interfaces/IUserRepository.cs]: Fixed

File: src/UserService.Business/UserCreateCommand.cs (hunk 6 above)
Comment (djino97) [reply src/UserService.Business/UserCreateCommand.cs]: Fixed

**Review by DarkFelas — APPROVED** [review]

**Pull request merged**